### PR TITLE
feat: user management

### PR DESCRIPTION
**Pull request by ferruhcihan**
### Implements
https://jira.linode.com/browse/APL-27
https://jira.linode.com/browse/APL-221
https://jira.linode.com/browse/APL-229

### Description
This PR adds user schema into the values-schema and updates the values files.

Is paired with: 
Console => https://github.com/linode/apl-console/pull/441
Api => https://github.com/linode/apl-api/pull/565
Tasks => https://github.com/linode/apl-tasks/pull/118
Docs => https://github.com/linode/apl-docs/pull/76

## Checklist

- [ ] Architecture Design Records have been added as `adr/*.md` and appended to list in `adr/_index.md`, if applicable.
- [ ] The `values-schema.yaml` file and `test/**` fixtures have been updated to reflect code changes, if applicable.
- [ ] The OpenApi Schema from redkubes/otomi-api project is compatible with definitions from `values-schema.yaml` file, if applicable.
- [ ] Helm releases are meeting otomi's baseline security policies, if applicable.
- [ ] Helm chart and helmfile changes are tested against upgrade scenario, if applicable.

### CODE REVIEW

File: values-schema.yaml [hunk 1]
```diff
@@ -1435,6 +1433,39 @@ definitions:
     required:
       - name
       - url
+  user:
+    type: object
+    description: A user can be a team member or an admin.
```
Comment (ElderMatt) [inline values-schema.yaml:1438]: Would it be good to mention that the user has to do with keycloak?

**Review by ElderMatt — CHANGES_REQUESTED** [review]

File: values-schema.yaml [hunk 2]
```diff
@@ -1435,6 +1433,39 @@ definitions:
...
+        x-secret: ''
+      isTeamAdmin:
+        type: boolean
+        x-secret: ''
+      teams:
+        type: array
+        items:
+          type: string
+          x-secret: ''
+      initialPassword:
```
Comment (j-zimnowoda) [inline values-schema.yaml:1460]: what will happen if I create a user without initialPassword ?

File: values-schema.yaml (hunk 1 above)
Comment (ferruhcihan) [reply values-schema.yaml]: Yes, that's right, that would be good. I added it in the description.

File: values/apl-keycloak-operator/apl-keycloak-operator-raw.gotmpl [hunk 3]
```diff
@@ -18,6 +18,8 @@
 {{- $k := $c | get "keycloak" }}
 {{- $doms := tpl (readFile "../../helmfile.d/snippets/domains.gotmpl") $v | fromYaml }}
 {{- $joinTpl := readFile "../../helmfile.d/utils/joinListWithSep.gotmpl" }}
+{{- $users := $v | get "users" list }}
```
[the remark this answers is not in the document]
Comment (ferruhcihan) [reply values/apl-keycloak-operator/apl-keycloak-operator-raw.gotmpl]: Good to learn this. I added users to defaults.yaml, updated this line and tested it. It works.

File: tests/bootstrap/input.yaml [hunk 4]
```diff
@@ -7,3 +7,10 @@ otomi:
 apps:
   metrics-server:
     enabled: false
+users:
```
[the remark this answers is not in the document]
Comment (ferruhcihan) [reply tests/bootstrap/input.yaml:10]: After adding users to defaults.yaml, I could bootstrap values with users and could see the env/secrets.users.yaml file.

File: values-schema.yaml (hunk 2 above)
Comment (ferruhcihan) [reply values-schema.yaml:1460]: I tested this issue. If we create a user without initialPassword, the apl-keycloak-operator will create a user with an `undefined` (as string) initial password (which is not good). We should handle this in the apl-keycloak-operator and generate an initial random password before the user creation but that initial password will not stored anywhere so the user needs to reset password by platform admins. Do you have any idea? @j-zimnowoda @srodenhuis @ElderMatt

File: values-schema.yaml [hunk 5]
```diff
@@ -2887,7 +2918,9 @@ properties:
       clientSecret:
         type: string
         x-secret: ''
-      adminGroupID:
+      platformAdminGroupID:
+        $ref: '#/definitions/wordCharacterPattern'
+      allTeamsAdminGroupID:
```
[the remark this answers is not in the document]
Comment (ferruhcihan) [reply values-schema.yaml:2923]: Added lines in values-changes.yaml file for user management values migration.

File: values-schema.yaml (hunk 2 above)
Comment (j-zimnowoda) [reply values-schema.yaml:1460]: Can bootstrap.ts do it?

File: values-schema.yaml (hunk 5 above)
[the remark this answers is not in the document]
Comment (j-zimnowoda) [reply values-schema.yaml:2923]: thx

File: tests/bootstrap/input.yaml (hunk 4 above)
[the remark this answers is not in the document]
Comment (j-zimnowoda) [reply tests/bootstrap/input.yaml:10]: confirmed. thx

File: values-schema.yaml (hunk 2 above)
Comment (ferruhcihan) [reply values-schema.yaml:1460]: Yes, in `bootstrap.ts` we can generate a random password and add it in the users who do not have an initial password.

Comment (ferruhcihan) [reply values-schema.yaml:1460]: `bootstrap.ts` will generate random initial passwords for users if they don't have one.

File: values/apl-keycloak-operator/apl-keycloak-operator-raw.gotmpl [hunk 6]
```diff
@@ -18,7 +18,14 @@
 {{- $k := $c | get "keycloak" }}
 {{- $doms := tpl (readFile "../../helmfile.d/snippets/domains.gotmpl") $v | fromYaml }}
 {{- $joinTpl := readFile "../../helmfile.d/utils/joinListWithSep.gotmpl" }}
-{{- $users := $v.users }}
+{{ $users := list }}
+{{- range $user := $v.users }}
```
Comment (j-zimnowoda) [inline values/apl-keycloak-operator/apl-keycloak-operator-raw.gotmpl:22]: Fantastic! I can see now, what groups I can expect in Keycloak.

File: values-schema.yaml (hunk 2 above)
Comment (j-zimnowoda) [reply values-schema.yaml:1460]: Nice, I see passwords being generated now.

**Review by j-zimnowoda — APPROVED** [review]
Great job!

**Review by ElderMatt — APPROVED** [review]

**Pull request merged**